crates/nu-command/src/progress_bar.rs

```diff
@@ -0,0 +1,71 @@
+use indicatif::{ProgressBar, ProgressState, ProgressStyle};
+use std::fmt;
+
+// This module includes the progress bar used to show the progress when using the command `save`
+// Eventually it would be nice to find a better plece for it.
+
+pub struct NuProgressBar {
+    pub pb: ProgressBar,
+    bytes_downloaded: u64,
+    total_bytes: Option<u64>,
+}
+
+impl NuProgressBar {
+    pub fn new(total_bytes: Option<u64>) -> NuProgressBar {
+        match total_bytes {
+            Some(total_size) => NuProgressBar {
+                pb: ProgressBar::new(total_size),
+                total_bytes: total_bytes,
+                bytes_downloaded: 0,
+            },
+            _ => {
+                let r_num = 10; // <== the value of this variable doesn't matter
+                NuProgressBar {
+                    pb: ProgressBar::new(r_num),

```

Comment:
Most GUI frameworks I've used have an "indeterminate" spinner / progress bar that can be used when calculating percentage finished is not possible. Does `indicatif` not have something like that?